### PR TITLE
added an email option to multiple trial upload

### LINKED ISSUE #4729: Implement email notifications for uploading large number of trials into the Database.
Expected Behavior 
--------------------------------------------------------------------------

Currently, we don't get an email notification about the status of uploading large numbers of trials using GUI.

For Bugs:
---------

### PR DESCRIPTION
**Pull request by Sri-2023**
Description 
-----------------------------------------------------

this will fix #4729 

Checklist 
---------------------------------------------------------------------------------------
- [ ] Refactoring only
- [ ] Documentation only
- [ ] Fixture update only
- [ ] Bug fix
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [ ] New feature
  - [ ] Relevant tests have been created and run.
  - [ ] Data was added to the fixture
    - [ ] Data was added via a patch in `/t/data/fixture/patches/`.
  - [ ] User-Facing Change
    - [ ] The user manual in `/docs` has been updated.
  - [ ] Any new Perl has been documented using **perldoc**.
  - [ ] Any new JavaScript has been documented using **JSDoc**.
  - [ ] Any new _legacy_ JavaScript has been moved from `/js` to `/js/source/legacy`.

### CODE REVIEW

File: lib/SGN/Controller/AJAX/Trial.pm [hunk 1]
```diff
@@ -1247,115 +1269,137 @@ sub upload_multiple_trial_designs_file_POST : Args(0) {
...
-        print STDERR "Transaction Error: $_\n";
-        push @{$save{'errors'}}, $_;
-    };
+    # print STDERR "breeding_program_name:  $breeding_program_name \n";
+    # print STDERR "infile:  $infile \n";
+
+    my $async_upload = CXGN::Tools::Run->new();
+    $async_upload->run_async("perl $basepath/bin/upload_multiple_trial_design.pl -H $dbhost -D $dbname -P $dbpass -w $basepath -U $dbuser -b $breeding_program_name -i $infile -un $username -e $email_address -eo $email_option_enabled -r $temp_file_nd_experiment_id");
+
+    # my $coderef = sub {
```
Comment (nmenda) [inline lib/SGN/Controller/AJAX/Trial.pm:1299]: where was this code moved? if it shouldn't be here then I'd remove it instead of commenting out

Comment (Sri-2023) [reply lib/SGN/Controller/AJAX/Trial.pm]: @nmenda, thanks for the remainder. Sorry, it was even mentioned in previous meetings, and missed it. Now it is fixed.

**Review by lukasmueller — COMMENTED** [review]
is an email being sent? I see no trace of it... may be due to the config of the test docker?

**Comment by Sri-2023** [comment]
> is an email being sent? I see no trace of it... may be due to the config of the test docker?

Yes, it will send an email if it is successful, and also if it finds errors. You will see something like this in your inbox. 
And also, see on the error log,  when the email is sent.

image
Screenshot from 2024-08-01 10-50-57

**Review by nmenda — APPROVED** [review]

**Review by lukasmueller — APPROVED** [review]
:-)

**Pull request merged**